### PR TITLE
feat: LemonSelect updates

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

Using the `LemonSelect` requires a complicated `LemonSelectOptions` structure which requires typing `as Blah` too often

## Changes

* Simplifies the generics used and provides only `LemonSelectSection[]` or `LemonSelectOption[]` as allowed options props. Internally boxes provided `LemonSelectOption[]` to a `LemonSelectSection` with no title. So the component is always processing `LemonSelectSection`s and doesn't have to check
* adds a max-height to the popup component as long popups could go unrecoverably offscreen
* to limit size of PR doesn't remove nested select from `AttributionFilter` or try to replace `CohortFields` with `LemonSelect`
* renames `LemonSelectOption::key` to be `LemonSelectOption::value` to more closely align with html select

#### before

broken-long-popup

#### after

fixed-long-props

## How did you test this code?

running it locally and seeing things work

### CODE REVIEW

File: frontend/src/lib/components/LemonSelect.tsx [hunk 1]
```diff
@@ -6,24 +6,23 @@ import { PopupProps } from './Popup/Popup'
 import './LemonSelect.scss'
 import clsx from 'clsx'
 
-export interface LemonSelectOption {
+export interface LemonSelectOption<T> {
+    key: T
```
Comment (benjackwhite) [inline frontend/src/lib/components/LemonSelect.tsx:10]: I know this is WIP but I definitely agreed in our conversation that we should rename this to `value` to be closer to the native select options and not conflict wit React "keys"

Comment (pauldambra) [reply frontend/src/lib/components/LemonSelect.tsx:10]: Ah, yes, good reminder! 📈

File: frontend/src/lib/components/LemonSelect.tsx [hunk 2]
```diff
@@ -109,32 +112,32 @@ export function LemonSelect<O extends LemonSelectOptions>({
...
+                                        if (option.value != localValue) {
+                                            onChange?.(option.value)
+                                            setLocalValue(option.value)
                                         }
                                     }}
                                     status="stealth"
                                     /* Intentionally == instead of === because JS treats object number keys as strings, */
                                     /* messing comparisons up a bit */
-                                    active={key == localValue}
+                                    active={option.value == localValue}
```
Comment (benjackwhite) [inline frontend/src/lib/components/LemonSelect.tsx:135]: I wonder... now that we aren't using object keys, can we remove the double equals?

**Review by benjackwhite — APPROVED** [review]
One comment of something that might be simpler now but otherwise - amaaazing 👍

File: frontend/src/lib/components/LemonSelect.tsx (hunk 2 above)
Comment (pauldambra) [reply frontend/src/lib/components/LemonSelect.tsx]: we can \o/

**Pull request merged**